Allow relative URLs in broadcasts action links

davelopez opened the pull request:
Fixes #17033

It relaxes the schema typing in notifications a bit to support relative URLs. The pydantic type `AnyUrl` always confuses me and makes me think it supports both absolute and relative URLs but it's not the case.

## How to test the changes?
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

davelopez committed:
Allow relative action URLs in broadcasts

davelopez committed:
Add test for broadcasts action links

mvdbeek commented on lib/galaxy/schema/types.py:
```diff
@@ -1,12 +1,16 @@
 # Making them an alias of `str` for now
 RelativeUrl = str
 
+AbsoluteOrRelativeUrl = Union[AnyUrl, RelativeUrl]
```
```suggestion
AbsoluteOrRelativeUrl = RelativeUrl
```

davelopez replied:
I was hoping this was validating the URL when it was absolute and then fallback to relative (str) when it failed... but I guess it will always fallback to str so maybe we can just remove `AbsoluteOrRelativeUrl` and use directly `RelativeUrl`?

mvdbeek replied:
Right, that is what happens, but we don't know that and it'll never catch anything ... htt:/url.com is going to be a valid string.

davelopez committed:
Keep AbsoluteOrRelativeUrl as RelativeUrl for now

This will always fall back to a string, but the intention of the code reads better and we may add some custom validation later to really check for the URL correctness.

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>
